packages/Form/Input/switch/src/Switch.js

```diff
@@ -20,72 +20,106 @@ const defaultProps = {
 };
 
 const SwitchItems = props => {
-  const { options, value, name, onChange, onBlur, onFocus, disabled, ...otherProps } = props;
+  const {
+    options,
+    value,
+    name,
+    onChange,
+    onBlur,
+    onFocus,
+    disabled,
+    ...otherProps
+  } = props;
   return options.map(option => {
     const isChecked = option.value === value;
-    return (<SwitchItem
-      key={option.value}
-      id={option.id}
-      value={option.value}
-      label={option.label}
-      isChecked={isChecked}
-      name={name}
-      onChange={onChange}
-      onBlur={onBlur}
-      onFocus={onFocus}
-      disabled={disabled}
-      {...omitProperties(otherProps)}
-    />)
-  })
+    return (
+      <SwitchItem
+        key={option.value}
+        id={option.id}
+        value={option.value}
+        label={option.label}
+        isChecked={isChecked}
+        name={name}
+        onChange={onChange}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        disabled={disabled}
+        {...omitProperties(otherProps)}
+      />
+    );
+  });
 };
 
-export const getSliderPosition = (value, options, ref) => {
-  const option = options.find(v => v.value === value);
-  const emptyResult = { left: 0, width: 0 };
-  if (!option || !ref.current) {
-    return emptyResult;
-  }
-  const currentElementResult = ref.current.querySelectorAll(`input[value="${value}"]`);
+export const getSliderPosition = (radioName, ref) => {
+  const currentElementResult =
+    ref.current &&
+    ref.current.querySelectorAll(`input[name="${radioName}"]:checked`);
   if (!currentElementResult || currentElementResult.length !== 1) {
-    return emptyResult;
+    return { left: 0, width: 0 };
   }
   const currentElement = currentElementResult[0].parentNode;
   const left = currentElement.offsetLeft;
   const width = currentElement.clientWidth;
   return {
     left,
-    width
+    width,
   };
 };
 
 class Switch extends React.Component {
   constructor(props) {
     super(props);
     this.ref = React.createRef();
+    this.state = {
+      selectedValue: props.value,
+      sliderStyle: {
+        display: 'none',
+      },
+    };
   }
 
-  render() {
-    const { value, options } = this.props;
-    const sliderPosition = getSliderPosition(value, options, this.ref);
-    const sliderStyle = {
+  componentDidMount() {
+    this.setState({
+      sliderStyle: this.getSliderStyle(),
+    });
+  }
+
+  getSliderStyle() {
```

Comment:
Why it’s not just a className that adjust the style?

Reply:
Because style must be calculate after 1st render & on each rerender